### PR TITLE
Add rich property editors for design-time experience in WPF .NET Framework applications

### PR DESCRIPTION
**Pull request by DVaughan**
### Description of Change ###

Includes a new Design project for .NET Framework applications, which targets the Designer Isolation architecture.
NuGet package includes an Install.ps1 powershell script that removes the reference to the .Design.dll when adding package to .NET Framework projects. 

### API Changes ###

No public API changes.

### Behavioral Changes ###

Value editors now appear in the property grid when editing behavior properties, such as the SourceObject of the EventTrigger.

### CODE REVIEW

File: src/Microsoft.Xaml.Behaviors.Design/Microsoft.Xaml.Behaviors.Design.csproj [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+  <PropertyGroup>
+    <Configuration Condition=" '$(Configuration)' == '' ">Debug</Configuration>
+    <Platform Condition=" '$(Platform)' == '' ">AnyCPU</Platform>
+    <ProjectGuid>{6FBA03AC-D43F-447F-A087-404B9DB14952}</ProjectGuid>
+    <OutputType>Library</OutputType>
+    <AppDesignerFolder>Properties</AppDesignerFolder>
+    <RootNamespace>Microsoft.Xaml.Behaviors.Design</RootNamespace>
+    <AssemblyName>Microsoft.Xaml.Behaviors.Design</AssemblyName>
+    <TargetFrameworkVersion>v4.7.2</TargetFrameworkVersion>
+    <FileAlignment>512</FileAlignment>
```
Comment (mgoertz-msft) [inline src/Microsoft.Xaml.Behaviors.Design/Microsoft.Xaml.Behaviors.Design.csproj]: Shouldn't that be 4.5 to match the behaviors TFM?

File: src/Microsoft.Xaml.Behaviors.DesignTools/Properties/Resources.Designer.cs [hunk 2]
```diff
@@ -8,7 +8,7 @@
 // </auto-generated>
 //------------------------------------------------------------------------------
 
-namespace Microsoft.Xaml.Behaviors.DesignTools.Properties {
+namespace Microsoft.Xaml.Behaviors.DesignTools {
     using System;
```
Comment (mgoertz-msft) [inline src/Microsoft.Xaml.Behaviors.DesignTools/Properties/Resources.Designer.cs]: That's surprising

File: src/Microsoft.Xaml.Behaviors.DesignTools/Microsoft.Xaml.Behaviors.DesignTools.csproj [hunk 3]
```diff
@@ -63,6 +62,7 @@
       <Generator>ResXFileCodeGenerator</Generator>
       <LastGenOutput>Resources.Designer.cs</LastGenOutput>
       <SubType>Designer</SubType>
+      <CustomToolNamespace>Microsoft.Xaml.Behaviors.DesignTools</CustomToolNamespace>
     </EmbeddedResource>
```
Comment (mgoertz-msft) [inline src/Microsoft.Xaml.Behaviors.DesignTools/Microsoft.Xaml.Behaviors.DesignTools.csproj]: Any particular reason to change this?

File: src/Microsoft.Xaml.Behaviors.Design/Properties/AssemblyInfo.cs [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+[assembly: AssemblyCulture("")]
+
+// Setting ComVisible to false makes the types in this assembly not visible
+// to COM components.  If you need to access a type in this assembly from
+// COM, set the ComVisible attribute to true on that type.
+[assembly: ComVisible(false)]
+
+// The following GUID is for the ID of the typelib if this project is exposed to COM
+[assembly: Guid("6fba03ac-d43f-447f-a087-404b9db14952")]
+
```
Comment (mgoertz-msft) [inline src/Microsoft.Xaml.Behaviors.Design/Properties/AssemblyInfo.cs]: Why would we want this here if ComVisible is false and we don't want anyone to reference this anyway?

File: src/Microsoft.Xaml.Behaviors.DesignTools/Microsoft.Xaml.Behaviors.DesignTools.csproj (hunk 3 above)
Comment (DVaughan) [reply src/Microsoft.Xaml.Behaviors.DesignTools/Microsoft.Xaml.Behaviors.DesignTools.csproj]: It was made the same across the two projects.
Having it at the root eliminates the need for a using statement.

File: src/Microsoft.Xaml.Behaviors.DesignTools/Properties/Resources.Designer.cs (hunk 2 above)
Comment (DVaughan) [reply src/Microsoft.Xaml.Behaviors.DesignTools/Properties/Resources.Designer.cs]: Please see the comment below.

File: src/Microsoft.Xaml.Behaviors/Install.ps1 [hunk 5]
```diff
@@ -0,0 +1,5 @@
+param($installPath, $toolsPath, $package, $project)
+
+# Remove the reference to the .Design.dll, which is incorrectly referenced during
+# the NuGet package installation in .NET Framework applications (not .NET Core).
+$project.Object.References | Where-Object { $_.Name -eq 'Microsoft.Xaml.Behaviors.Design' } | ForEach-Object { $_.Remove() }
```
Comment (nkolev92) [inline src/Microsoft.Xaml.Behaviors/Install.ps1]: Make sure you test the uninstall scenario of this package in .NET Framework packages.config projects.

File: src/Microsoft.Xaml.Behaviors.Design/Properties/AssemblyInfo.cs (hunk 4 above)
Comment (DVaughan) [reply src/Microsoft.Xaml.Behaviors.Design/Properties/AssemblyInfo.cs]: Removed

File: src/Microsoft.Xaml.Behaviors.DesignTools/Microsoft.Xaml.Behaviors.DesignTools.csproj (hunk 3 above)
Comment (mgoertz-msft) [reply src/Microsoft.Xaml.Behaviors.DesignTools/Microsoft.Xaml.Behaviors.DesignTools.csproj]: And added the need for a custom tool namespace...  ;-)

Anyway, it's fine I was just curious.

---
In reply to: [349328568](https://github.com/microsoft/XamlBehaviorsWpf/pull/60#discussion_r349328568) [](ancestors = 349328568)

File: src/Microsoft.Xaml.Behaviors/Install.ps1 (hunk 5 above)
Comment (DVaughan) [reply src/Microsoft.Xaml.Behaviors/Install.ps1]: Works as expected.

File: src/Microsoft.Xaml.Behaviors.Design/Microsoft.Xaml.Behaviors.Design.csproj (hunk 1 above)
Comment (DVaughan) [reply src/Microsoft.Xaml.Behaviors.Design/Microsoft.Xaml.Behaviors.Design.csproj]: Changed

**Pull request merged**